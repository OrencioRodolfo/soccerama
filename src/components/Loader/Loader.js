import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { LinearProgress } from 'material-ui/Progress'
import styled from 'styled-components'

const LoaderWrapper = styled.div`
  position: fixed;
  width: 100%;
`

const Loader = ({ show }) => {
  if (show) {
    return (
      <LoaderWrapper>
        <LinearProgress />
      </LoaderWrapper>
    )
  }

  return null
}

Loader.propTypes = {
  show: PropTypes.bool,
}

Loader.defaultProps = {
  show: false,
}

const mapStateToProps = ({ api }) => ({
  show: api.pendingRequests,
})

export default connect(mapStateToProps)(Loader)
export {
  Loader as DLoader,
  mapStateToProps,
}
