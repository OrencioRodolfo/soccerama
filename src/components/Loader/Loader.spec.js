import React from 'react'
import { shallow } from 'enzyme'
import { DLoader, mapStateToProps } from './Loader'

describe('Loader component', () => {
  it('Must render nothing', () => {
    const component = shallow(<DLoader />)
    expect(component).toMatchSnapshot()
  })

  it('Must render properly', () => {
    const component = shallow(<DLoader show />)
    expect(component).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('must return an object with the "show" property', () => {
      const result = mapStateToProps({ api: { pendingRequests: true } })
      expect(result).toEqual({ show: true })
    })
  })
})
