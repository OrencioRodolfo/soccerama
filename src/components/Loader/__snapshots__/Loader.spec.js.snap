// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component Must render nothing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader
    show={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Loader component Must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader
    show={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(LinearProgress) />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(LinearProgress) />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
